### PR TITLE
Azure Function App template

### PR DESCRIPTION
**Pull request by likelymichael**
Waiting on module to merge to update SHA before merging this

### CODE REVIEW

File: azure-function-app/massdriver.yaml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                - subdomain
+            - properties:
+                enable_dns:
+                  const: false
+
+connections:
+  required:
+    - azure_service_principal
+    - azure_virtual_network
+  # <md- range $key, $art:= .Connections md>
```
Comment (coryodaniel) [inline azure-function-app/massdriver.yaml]: Are we going to uncomment before publishing?

Comment (likelymichael) [reply azure-function-app/massdriver.yaml]: Ah, yes, sorry. Commented out for local testing.

File: azure-function-app/massdriver.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+name: <md .Name md>
+description: <md .Description md>
+source_url: github.com/YOUR_ORG/<md .Name md>
+access: private
+type: application
+
+app:
+  envs:
+    # Use jq expressions to build environment variables from input params or connections
+    LOG_LEVEL: .params.log_level
```
Comment (coryodaniel) [inline azure-function-app/massdriver.yaml:11]: We should have a param ref’d in the example (that’s uncommented) that doesn’t actually exist. It’ll probably fail during deployment t. Saw this in the k8s examples the other day.

Comment (likelymichael) [reply azure-function-app/massdriver.yaml:11]: Sure. Is there an example I can follow? Looked at both k8s example apps and I'm not sure how that would translate here.

Comment (coryodaniel) [reply azure-function-app/massdriver.yaml:11]: I’d say add a param as an example w enum “info” “warn” “error” and comment both out. The user then has a working example they can uncomment if it applies.

Comment (likelymichael) [reply azure-function-app/massdriver.yaml:11]: Should this be added to the function app example or the function app template?

Comment (coryodaniel) [reply azure-function-app/massdriver.yaml:11]: Template. It does the jq expression eval

Comment (likelymichael) [reply azure-function-app/massdriver.yaml:11]: Something like this?
```
    LOG_LEVEL: # .params.log_level
      # type: string
      # enum:
      #   - info
      #   - warn
      #   - error
```

File: azure-function-app/massdriver.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      application:
+        sku_name: P3v3
+        minimum_worker_count: 2
+        maximum_worker_count: 30
+  required:
+    - application
+    - docker
+    - dns
+  properties:
+    application:
```
Comment (WillBeebe) [inline azure-function-app/massdriver.yaml:140]:
```suggestion
    # As an example, uncomment the section below to add a parameter for the logging level of your application. See the app.envs section above for how to set environment values from application parameters.
    # log_level:
    #   type: string
    #   title: Log Level
    #   enum: 
    #   - info 
    #   - warn 
    #   - error
    application: 
```

File: azure-function-app/massdriver.yaml (hunk 2 above)
Comment (WillBeebe) [inline azure-function-app/massdriver.yaml:11]:
```suggestion
    # Use jq expressions to map parameters to environment variables. You can find the parameter definition below in the params section
    # LOG_LEVEL: .params.log_level
```

File: azure-function-app/massdriver.yaml [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+      # enum:
+      #   - info
+      #   - warn
+      #   - error
+    # MONGO_DSN: .connections.mongo_authentication.data.authentication.username + ":" + .connections.mongo_authentication.data.authentication.password + "@" + .connections.mongo_authentication.data.authentication.hostname + ":" + (.connections.mongo_authentication.data.authentication.port|tostring)
+  policies: []
+    # Use jq expressions to select policies from artifact security blocks
+    # - .connections.sqs.data.security.iam.subscribe
+
+MonitorAlarmMetric: &monitor_alarm_metric
```
Comment (WillBeebe) [inline azure-function-app/massdriver.yaml:17]: This is pretty sweet

**Pull request merged**